gno.land/pkg/gnoland/mock_test.go

```diff
@@ -45,21 +45,27 @@ func (m *mockEventSwitch) RemoveListener(listenerID string) {
 	}
 }
 
-type (
-	addPackageDelegate func(sdk.Context, vm.MsgAddPackage) error
-	callDelegate       func(sdk.Context, vm.MsgCall) (string, error)
-	queryEvalDelegate  func(sdk.Context, string, string) (string, error)
-	runDelegate        func(sdk.Context, vm.MsgRun) (string, error)
-)
-
 type mockVMKeeper struct {
-	addPackageFn addPackageDelegate
-	callFn       callDelegate
-	queryFn      queryEvalDelegate
-	runFn        runDelegate
+	addPackageFn                func(sdk.Context, vm.MsgAddPackage) error
+	callFn                      func(sdk.Context, vm.MsgCall) (string, error)
+	queryFn                     func(sdk.Context, string, string) (string, error)
+	runFn                       func(sdk.Context, vm.MsgRun) (string, error)
+	loadStdlibFn                func(sdk.Context, string)
+	loadStdlibCachedFn          func(sdk.Context, string)
+	makeGnoTransactionStoreFn   func(ctx sdk.Context) sdk.Context
+	commitGnoTransactionStoreFn func(ctx sdk.Context)
+
+	calls map[string]int
+}
+
+func (m *mockVMKeeper) call(name string) {
+	if m.calls != nil {
+		m.calls[name]++
+	}
 }
 
 func (m *mockVMKeeper) AddPackage(ctx sdk.Context, msg vm.MsgAddPackage) error {
+	m.call("AddPackage")
```

Comment:
I'm a very much against this type of testing. What does this information give you, that writing a unit test mock handler doesn't?

Please remove this, I think it's excessive, and just introduces more maintenance on the mock

Reply:
I don't believe it is.

https://github.com/gnolang/gno/blob/40fae4827aef89e22d740f570a64054376f95cb5/gno.land/pkg/gnoland/app_test.go#L159-L168

If we don't check the number of calls happening within InitChainer at that point, we have no way to tell whether makeGnoTransactionStore and LoadStdlib were properly called by the InitChainer; ie. it's useless to have assertions inside of them if you don't also test whether they've been called, too.

Reply:
I'm willing to have this as simple integer counters within that test, if you wish

Reply:
It's a nitpick, but I think it's a better approach -- you control the mock in the unit test so this data can be captured